### PR TITLE
12202 improve current deliverables price calculation/display

### PR DESCRIPTION
**Pull request by cialuo**
Business requirement deliverable 8.0  12202 improve current deliverables price calculation/display

### CODE REVIEW

File: business_requirement_deliverable/models/business.py [hunk 1]
```diff
@@ -141,10 +141,38 @@ def product_id_change(self):
             description = product.name
             uom_id = product.uom_id.id
             unit_price = product.list_price
+        if self.business_requirement_id.project_id.pricelist_id and \
```
Comment (elicoidal) [inline business_requirement_deliverable/models/business.py]: There is not pricelist in project AFAIK.
Anyway it needs to be a separate pricelist for estimation

File: business_requirement_deliverable/models/business.py [hunk 2]
```diff
@@ -141,10 +141,38 @@ def product_id_change(self):
...
+                self.business_requirement_id.partner_id and self.uom_id:
+            product = self.product_id.with_context(
+                lang=self.business_requirement_id.partner_id.lang,
+                partner=self.business_requirement_id.partner_id.id,
+                quantity=self.qty,
+                pricelist=self.business_requirement_id.
+                project_id.pricelist_id.id,
+                uom=self.uom_id.id,
+            )
+            unit_price = product.price
```
Comment (elicoidal) [inline business_requirement_deliverable/models/business.py:154]: Does this calculate the price according to the provided pricelist?

File: business_requirement_deliverable/models/project.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# © 2016 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import fields, models
+
+
+class Project(models.Model):
+    _inherit = "project.project"
+
+    pricelist_id = fields.Many2one(
+        string='Pricelist',
```
Comment (elicoidal) [inline business_requirement_deliverable/models/project.py]: "Estimation Pricelist"
help: "Pricelist used for the estimation of the Business Requirements Deliverables linked to this project. Currency of the Deliverables will be the one from this pricelist."

**Comment by elicoidal** [comment]
some details other than that, LGTM

File: business_requirement_deliverable/models/business.py (hunk 2 above)
Comment (cialuo) [inline business_requirement_deliverable/models/business.py:154]: Yes, it will

**Comment by duanyp1991** [comment]
LGTM

**Pull request merged**